components/tidb_query_datatype/src/codec/mysql/set.rs

```diff
@@ -100,6 +100,12 @@ impl<'a> SetRef<'a> {
     pub fn new(data: &'a BufferVec, value: u64) -> Self {
         Self { data, value }
     }
+    pub fn to_owned(self) -> Set {
+        Set {
+            data: Arc::new(self.data.clone()),
```

Comment:
Where do we use `to_owned`? If we use this function too much, I would suggest converting `self.data` to an Arc and just clone `Arc` here.

Reply:
Required by `ScalarValue`'s `to_owned`, which only be used here:

https://github.com/tikv/tikv/blob/73f01599d7dab385b532b94f69f1bac6d18c1198/components/tidb_query_executors/src/stream_aggr_executor.rs#L298-L302

Looks not so much.

Reply:
If cloning every key requires cloning its corresponding enum vector, I think there would be performance regression :(

Reply:
Let's convet `EnumRef` / `SetRef`'s data to a Arc here.

Reply:
> Let's convet EnumRef / SetRef's data to a Arc here.

If we change like this, `EnumRef` is the same with `Enum`, any ideas about that?

Reply:
I got that point. Let's change back and see if we could optimize the executor not to clone later, instead of reducing the cost of clone in `Set` and `Enum`. If we save `Arc` in `EnumRef`, then constructing every `Ref` would lead to an operation on atomic reference counter.